release_dashboard/lib/widgets/create_release_substeps.dart

```diff
@@ -112,30 +153,54 @@ class CreateReleaseSubstepsState extends State<CreateReleaseSubsteps> {
 }
 
 typedef SetReleaseData = void Function(String name, String data);
+typedef ChangeIsEachInputValid = void Function(String name, bool isValid);
 
 /// Captures the input values and updates the corresponding field in [releaseData].
 class InputAsSubstep extends StatelessWidget {
-  const InputAsSubstep({
+  InputAsSubstep({
     Key? key,
-    required this.index,
+    required this.substepName,
     required this.setReleaseData,
     this.hintText,
+    required this.changeIsInputValid,
   }) : super(key: key);
 
-  final int index;
+  final String substepName;
   final SetReleaseData setReleaseData;
   final String? hintText;
+  final ChangeIsEachInputValid changeIsInputValid;
 
   @override
   Widget build(BuildContext context) {
+    late RegExp formRegexValidator;
+    late String validatorErrorMsg;
+
+    Map<String, Object> RegexAndErrorMsg = git(name: substepName).getRegexAndErrorMsg();
+    formRegexValidator = RegexAndErrorMsg['regex'] as RegExp;
+    validatorErrorMsg = RegexAndErrorMsg['errorMsg'] as String;
+
     return TextFormField(
-      key: Key(CreateReleaseSubsteps.substepTitles[index]),
+      key: Key(substepName),
+      autovalidateMode: AutovalidateMode.onUserInteraction,
       decoration: InputDecoration(
-        labelText: CreateReleaseSubsteps.substepTitles[index],
+        labelText: substepName,
         hintText: hintText,
       ),
       onChanged: (String data) {
-        setReleaseData(CreateReleaseSubsteps.substepTitles[index], data);
+        data = data.trim();
```

Comment:
If users can enter multiple git hashes, we may want to do `data.replaceAll(' ', '')` to remove the whitespace (trim only removes from the ends)